content/chainguard/chainguard-images/features/ca-docs/custom-assembly-api-demo.md

```diff
@@ -0,0 +1,299 @@
+menu:
+  docs:
+    parent: "features"
+weight: 015
+toc: true
+---
+
+Chainguard's [Custom Assembly](/chainguard/chainguard-images/features/custom-assembly/) is a tool that allows customers to create customized containers with extra packages added. This enables customers to reduce their risk exposure by creating container images that are tailored to their internal organization and application requirements while still having few-to-zero CVEs.
+
+You can use the Chainguard API to further customize your Custom Assembly builds and retrieve information about them. This tutorial highlights a demo application (which can be found in [Chainguard Academy's Demo Applications repository](https://github.com/chainguard-dev/edu-images-demos/tree/main)) which, when run, updates a Custom Assembly container's configuration based on a provided YAML file.
+
+
+## Prerequisites
+
+In order to follow along with this guide, you will need the following:
+
+* Access to a Custom Assembly container image. If your organization doesn't yet have access to Custom Assembly, reach out to your account team to start the process.
+* The demo application used in this guide is written in Go, so you will need Go installed on your local machine. Refer to the [official Go documentation](https://go.dev/doc/install) for instructions on downloading and installing Go.
+* You will also need [`chainctl` installed](/chainguard/chainctl-usage/how-to-install-chainctl/) on your local machine to create a Chainguard token and authenticate to the Chainguard API.
+
+
+## Downloading the Demo Application
+
+This step involves downloading the demo application code to your local machine. To ensure that the application files don't remain on your system, navigate to a temporary directory like `/tmp/`:
+
+```shell
+cd /tmp/
+```
+
+Your system will automatically delete the `/tmp/` directory's contents the next time it shuts down or reboots.
+
+The code that comprises this demo application is hosted in a public GitHub repository managed by Chainguard. Pull down the example application files from GitHub with the following command:
+
+```shell
+git clone --sparse https://github.com/chainguard-dev/edu-images-demos.git
+```
+
+Because this guide's demo application code is stored in a repository with other examples, we don't need to pull down every file from this repository. For this reason, this command includes the `--sparse` option. This initializes a sparse-checkout file, causing the working directory to contain only the files in the root of the repository until the sparse-checkout configuration is modified.
+
+Navigate into this new directory and list its contents to confirm this:
+
+```shell
+cd edu-images-demos/ && ls
+```
+
+For now, this directory only contains the repository's `LICENSE` and `README` files:
+
+```
+LICENSE  README.md
+```
+
+To retrieve the files you need for this tutorial's sample application, run the following `git` command:
+
+```shell
+git sparse-checkout set custom-assembly-go
```

Comment:
I've used git since something like 2011 and this is the first time I've used sparse-checkout. Thanks for teaching me stuff while I'm focused on learning other stuff!